packages/date-picker/src/shared/components/DateInput/DateFormField/DateFormField.styles.ts

```diff
@@ -1,7 +1,26 @@
 import { css } from '@leafygreen-ui/emotion';
+import { Size } from '@leafygreen-ui/tokens';
 
-export const iconButtonStyles = css`
+export const calendarButtonSize: Record<Size, number> = {
+  [Size.XSmall]: 20,
+  [Size.Small]: 22,
+  [Size.Default]: 28,
+  [Size.Large]: 28,
+};
+
+// FormField does not account for icon width combined with the gap so this is a temp workaround
+const getMargin = (size: Size) => {
+  const iconWidthDiff = (calendarButtonSize[size] - 16) / 2; // spec is 16px width but eng width is wider because of hover/focus states
+  return -iconWidthDiff;
+};
+
+export const iconButtonStyles = (size: Size) => css`
```

Comment:
his could be a static object.
(Though it's debatable whether this is better. _Technically_ this function will be called on every render, but the perf impact is negligible)

```js
iconButtonStyles = {
 [Size.Small]: css`
    svg + button {
      margin-left: ${getMargin(Size.Small)}px;
    }
`
}
```